sklearn/tests/test_base.py

```diff
@@ -143,6 +145,24 @@ def test_clone_nan():
     assert_true(clf.empty is clf2.empty)
 
 
+def test_clone_sparse_matrices():
+    sparse_matrix_classes = [
+        getattr(sp, name)
+        for name in dir(sp) if name.endswith('_matrix')]
+
+    PY26 = sys.version_info[:2] == (2, 6)
+    if PY26:
+        # sp.dok_matrix can not be deepcopied in Python 2.6
```

Comment:
Yes, I think it's because Py2.6 has a specialised reconstruction routine for dicts and fails to handle the subclassing correctly... I'm happy with this solution.